### ISSUE TITLE
Creating spot instances in AWS result in error

### ISSUE THREAD

**Issue by cmd-ntrf** [ComputeCanada/magic_castle#212]
```╷
│ Error: Error requesting spot instances: UnknownParameter: The parameter NetworkCardIndex is not recognized
│ 	status code: 400, request id: 526f7fff-2b1c-48e3-823d-4105a60ad7f3
│ 
│   with module.aws.aws_spot_instance_request.spot_instances["node1"],
│   on aws/infrastructure.tf line 115, in resource "aws_spot_instance_request" "spot_instances":
│  115: resource "aws_spot_instance_request" "spot_instances" {
│ 
╵
╷
│ Error: Error requesting spot instances: UnknownParameter: The parameter NetworkCardIndex is not recognized
│ 	status code: 400, request id: e93cb921-c94b-4ac3-a0ff-33f5e30ddb15
│ 
│   with module.aws.aws_spot_instance_request.spot_instances["node2"],
│   on aws/infrastructure.tf line 115, in resource "aws_spot_instance_request" "spot_instances":
│  115: resource "aws_spot_instance_request" "spot_instances" {
```

This appears to be the result of a change introduced in https://github.com/hashicorp/terraform-provider-aws/pull/24283.

It is unclear at this point if it is a regression in Terraform AWS provider or something we will have to live with in the long run.

The solution for now is to set `hashicorp/aws` provider version to v4.10.0.

**Comment by congbang-le** [comment]
It still happens in `hashicorp/aws` version v4.45.0. The only way is downgrading to `v4.10.0` like @cmd-ntrf suggested. Thank you for the solution.

**Comment by gabrielar** [comment]
I know this issue has been closed for a while now, but I do have a proper fix for this in the terraform repo. It is PR https://github.com/hashicorp/terraform-provider-aws/pull/38336 . But to get this merged I need a few thumb ups 👍  on the PR so it gets some attention from the HashiCorp team. 

Also I have raised an issue here: https://github.com/hashicorp/terraform-provider-aws/issues/38335 . A 👍  on the issue may also help.
